test/functional/user_tags_controller_test.rb

```diff
@@ -41,22 +41,41 @@ def setup
     UserSession.create(rusers(:bob))
     valid_tags = ['skill', 'gear', 'role', 'tool']
     valid_tags.each do |tag|
-      post :create, :type => tag, :value => 'tagvalue'
+      post :create, :id => rusers(:bob).username, :type => tag, :value => 'tagvalue'
       assert_equal "tagvalue tag created successfully", flash[:notice]
       assert_redirected_to info_path
     end
 
     invalid_tags = ['skills', 'abc', '123']
     invalid_tags.each do |tag|
-      post :create, :type => tag, :value => 'tagvalue'
+      post :create, :id => rusers(:bob).username, :type => tag, :value => 'tagvalue'
       assert_equal "Error: Invalid value #{tag}", assigns['output']['errors'][0]
     end
   end
 
   test "should not allow empty tag value" do
     UserSession.create(rusers(:bob))
-    post :create, :type => "skill", :value => ''
+    post :create, :id => rusers(:bob).username, :type => "skill", :value => ''
     assert_equal "Error: value cannot be empty", assigns['output']['errors'][0]
     assert_redirected_to info_path
   end
+
+  test "admin should delete existing tag of normal user" do
+    UserSession.create(rusers(:bob))
+    post :create, :id => rusers(:bob).username, :type => "role", :value => "Organizer"
+    user_tag = UserTag.where(uid: rusers(:bob).id).last
+    user_tags_count = UserTag.where(uid: rusers(:bob).id).count
+    # Delete above tag
+    UserSession.create(rusers(:jeff))
+    post :delete, :id => user_tag.id
+    assert_equal user_tags_count -1, UserTag.where(uid: rusers(:bob).id).count
+  end
+
+  test "admin should create new tags for normal user" do
```

Comment:
Can you counter-test this, by confirming that *non-admins* can NOT create tags for others, but rename the ones above to show that this IS possible for your own tags? Thanks!!